pipeline/tests/mock_backend.rs

```diff
@@ -0,0 +1,76 @@
+use powdr_number::GoldilocksField;
+use powdr_pipeline::test_util::{make_simple_prepared_pipeline, test_mock_backend};
+
+fn col<const N: usize>(name: &str, values: [u64; N]) -> (String, Vec<GoldilocksField>) {
+    (
+        name.to_string(),
+        values.iter().map(|&x| GoldilocksField::from(x)).collect(),
+    )
+}
+
+fn init_logger() {
+    env_logger::builder().is_test(true).init();
+}
+
+#[test]
+fn fibonacci_correct_witness() {
+    init_logger();
+    let f = "pil/fibonacci.pil";
+    let pipeline = make_simple_prepared_pipeline::<GoldilocksField>(f);
+    let pipeline = pipeline.set_witness(vec![
+        col("Fibonacci::x", [1, 1, 2, 3]),
+        col("Fibonacci::y", [1, 2, 3, 5]),
+    ]);
+    test_mock_backend(pipeline);
+}
+
+#[test]
+#[should_panic(expected = "Constraint check failed")]
+fn fibonacci_wrong_initialization() {
+    // Initializes y with 2 instead of 1
+    // -> fails `ISLAST * (y' - 1) = 0;` in the last row
+    init_logger();
+    let f = "pil/fibonacci.pil";
+    let pipeline = make_simple_prepared_pipeline::<GoldilocksField>(f);
+    let pipeline = pipeline.set_witness(vec![
+        col("Fibonacci::x", [1, 2, 3, 5]),
+        col("Fibonacci::y", [2, 3, 5, 8]),
+    ]);
+    test_mock_backend(pipeline);
+}
+
+#[test]
+fn block_to_block_correct_witness() {
+    init_logger();
+    let f = "asm/block_to_block.asm";
```

Comment:
Don't we already test this in `asm.rs`?

Reply:
Yeah, true. Also for the correct fibonacci witness. Should I remove them? I think it's kinda nice to have them side by side.

Reply:
The other test tests more (more fields & provers). I removed the redundant tests here!